### ISSUE TITLE
Add all functionality needed to build CuckooSandbox machinery

### ISSUE THREAD

**Issue by ilyaglow** [ilyaglow/remote-virtualbox#2]
Reference: https://github.com/cuckoosandbox/cuckoo/blob/master/cuckoo/machinery/virtualbox.py
- [x] Connect to VirtualBox Webservice
- [x] Handle simple auth on Linux
- [x] List available VMs
- [x] Get VM's OS
- [x] Get VM's state
- [x] VM run
- [x] VM suspend
- [x] VM kill if suspend got too long
- [x] Restore to a specified snapshot
- [x] Dump PCAP
- [x] Dump memory
- [x] Exception handling

**Comment by ilyaglow** [comment]
It seems user can't manage snapshots using VirtualBox Webservice
```
VirtualBox error: rc=0x80004001 Method takeSnapshot is not implemented (0x80004001)
VirtualBox error: rc=0x80004001 Method restoreSnapshot is not implemented (0x80004001)
```

**Comment by ilyaglow** [comment]
Closed by e9aea907296514370f9aa9d5f0c7a6cacb9ea697
